File: problems/leetcode/7.cpp
#include <iostream>
#include <vector>
#include <stack>
#include <algorithm>
#include <string>

using std::vector;
using std::stack;
using std::string;

class Solution {
public:
    int reverse(int x) {
        int isNagetive = 1;
        if (x < 0) isNagetive = -1;
        string temp = std::to_string(x);
        for (int i = 0; i < temp.length() / 2; i++){
            char tempC = temp[i];
            temp[i] = temp[temp.length() - 1 - i];
            temp[temp.length() - 1 - i] = tempC;
        }
        long res = isNagetive;
        res *= std::stol(temp);
        if (res > INT_MAX || res < INT_MIN) return 0;
        return res;
    }
};

File: problems/leetcode/7.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            isNagetive = -1;
            x = -x;
        else:
            isNagetive = 1;
        res = 0;
        while x > 0:
            res *= 10;
            res += x % 10;
            x //= 10;
        res *= isNagetive;
        if res > 2 ** 31 - 1 or res < - 2 ** 31:
            return 0;
        else:
            return res;